__tests__/Unit/hooks/tasksApi.test.tsx

```diff
@@ -0,0 +1,212 @@
+import { setupServer } from 'msw/node';
+
+import handlers from '../../../__mocks__/handlers';
+import { PropsWithChildren } from 'react';
+import { act, renderHook } from '@testing-library/react-hooks';
+import { Provider } from 'react-redux';
+import { store } from '@/app/store';
+import {
+    useAddTaskMutation,
+    useGetAllTasksQuery,
+    useUpdateTaskMutation,
+} from '@/app/services/tasksApi';
+import { QueryStatus } from '@reduxjs/toolkit/dist/query';
+import {
+    failedAddNewTaskHandler,
+    failedGetTasks,
+    failedUpdateTaskHandler,
+} from '../../../__mocks__/handlers/tasks.handler';
+
+const server = setupServer(...handlers);
+
+beforeAll(() => {
+    server.listen();
+});
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+function Wrapper({
+    children,
+}: PropsWithChildren<Record<string, never>>): JSX.Element {
+    return <Provider store={store()}>{children}</Provider>;
+}
+
+describe('useGetAllTasksQuery()', () => {
+    test('returns all tasks', async () => {
+        const { result, waitForNextUpdate } = renderHook(
+            () => useGetAllTasksQuery(),
+            {
+                wrapper: Wrapper,
+            }
+        );
+        const initialResponse = result.current;
+        expect(initialResponse.data).toBeUndefined();
+        expect(initialResponse.isLoading).toBe(true);
+
+        await act(() => waitForNextUpdate());
+
+        const nextResponse = result.current;
+        const tasksData = nextResponse.data;
+        expect(tasksData).not.toBeUndefined();
+        expect(nextResponse.isLoading).toBe(false);
+        expect(nextResponse.isSuccess).toBe(true);
+    });
+
+    test('should fail to return all tasks with error', async () => {
+        server.use(failedGetTasks);
+        const { result, waitForNextUpdate } = renderHook(
+            () => useGetAllTasksQuery(),
+            {
+                wrapper: Wrapper,
+            }
+        );
+        const initialResponse = result.current;
+        expect(initialResponse.isLoading).toBe(true);
+        expect(initialResponse.data).toBeUndefined();
+
+        await act(() => waitForNextUpdate());
+
+        const nextResponse = result.current;
+        expect(nextResponse.isError).toBe(true);
+        expect(nextResponse.error).toHaveProperty('status', 500);
+    });
+});
+
+describe('useAddTaskMutation()', () => {
+    test('creates a new task', async () => {
+        const newTask = {
+            title: 'SSG and SSR config to be setup ',
+            type: 'feature',
+            status: 'AVAILABLE',
+            percentCompleted: 0,
+            priority: 'TBD',
+            github: {
+                issue: {
+                    status: 'open',
+                    id: 697061944,
+                },
+            },
+        };
```

Comment:
please use fixtures

Reply:
Have created an issue for this, shall address in subsequent PRs https://github.com/Real-Dev-Squad/website-status/issues/652